kernel/src/syscall/stat.rs

```diff
@@ -135,10 +193,151 @@ impl From<Metadata> for Stat {
     }
 }
 
-bitflags::bitflags! {
+/// Structures for the extended file attribute retrieval system call statx.
+#[derive(Debug, Clone, Copy, Pod, Default)]
+#[repr(C)]
+pub struct Statx {
+    /// What results were written
+    stx_mask: u32,
+    /// Preferred general I/O size
+    stx_blksize: u32,
+    /// Flags conveying information about the file
+    stx_attributes: u64,
+    /// Number of hard links
+    stx_nlink: u32,
+    /// User ID of owner
+    stx_uid: u32,
+    /// Group ID of owner
+    stx_gid: u32,
+    /// File mode
+    stx_mode: u16,
+    /// Padding
+    __spare0: [u16; 1],
+    /// Inode number
+    stx_ino: u64,
+    /// File size
+    stx_size: u64,
+    /// Number of 512-byte blocks allocated
+    stx_blocks: u64,
+    /// Mask to show what's supported in stx_attributes
+    stx_attributes_mask: u64,
+    /// Last access time
+    stx_atime: StatxTimestamp,
+    /// File creation time
+    stx_btime: StatxTimestamp,
+    /// Last attribute change time
+    stx_ctime: StatxTimestamp,
+    /// Last data modification time
+    stx_mtime: StatxTimestamp,
+    /// Device ID of special file (if bdev/cdev)
+    stx_rdev_major: u32,
+    stx_rdev_minor: u32,
+    /// ID of device containing file
+    stx_dev_major: u32,
+    stx_dev_minor: u32,
+    /// Mount ID
+    stx_mnt_id: u64,
+    /// Memory buffer alignment for direct I/O
+    stx_dio_mem_align: u32,
+    /// File offset alignment for direct I/O
+    stx_dio_offset_align: u32,
+    /// Spare space for future expansion
+    __spare3: [u64; 12],
+}
+
+impl From<Metadata> for Statx {
+    fn from(info: Metadata) -> Self {
+        let devid = DeviceId::from(info.dev);
+        let rdevid = DeviceId::from(info.rdev);
+        Self {
+            stx_mask: 0,
+            stx_blksize: info.blk_size as u32,
+            stx_attributes: 0,
+            stx_nlink: info.nlinks as u32,
+            stx_uid: info.uid.into(),
+            stx_gid: info.gid.into(),
+            stx_mode: info.type_ as u16 | info.mode.bits(),
+            __spare0: [0; 1],
+            stx_ino: info.ino,
+            stx_size: info.size as u64,
+            stx_blocks: (info.blocks * (info.blk_size / 512)) as u64,
+            stx_attributes_mask: 0,
+            stx_atime: StatxTimestamp::from(info.atime),
+            stx_btime: StatxTimestamp::from(info.atime),
+            stx_ctime: StatxTimestamp::from(info.ctime),
+            stx_mtime: StatxTimestamp::from(info.ctime),
+            stx_rdev_major: rdevid.major(),
+            stx_rdev_minor: rdevid.minor(),
+            stx_dev_major: devid.major(),
+            stx_dev_minor: devid.minor(),
+            stx_mnt_id: 0,
+            stx_dio_mem_align: 0,
+            stx_dio_offset_align: 0,
+            __spare3: [0; 12],
```

Comment:
We should add comments or `TODO`s for all the dummy implementations here. Someday when we will complete the implementation and these comments will be very helpful until then.